addons/test_mail/tests/test_mail_gateway.py

```diff
@@ -918,6 +919,65 @@ def test_message_route_alias_multiple_new(self):
         new_simple = self.env['mail.test.gateway'].search([('name', '=', 'Test Subject')])
         self.assertEqual(len(new_simple), 1, 'message_process: a new mail.test should have been created')
 
+    def test_message_route_alias_reply_author_notify(self):
+        """Make sure users are notified when a reply is sent to an alias address.
+
+        The alias owner should impact who the creator of the message is but not the notifications.
+        """
+
+        test_record = self.env['mail.test.ticket'].create({})
+        author_partner = self.env['res.partner'].create({
+            'name': self.email_from,
+            'email': self.email_from,
+        })
+        message = self.env['mail.message'].create({
+            'body': '<p>test</p>',
+            'email_from': 'reply_author_notify_test@test.com',
```

Comment:
Why has the message a from that is an alias ? This is quite not common normally ... is it related to the test ?

Reply:
email_from being the alias is relevant to cases where you set the alias to be your own email address, as a user. Such that e.g. whenever someone send an email to your user it creates a task for you to handle. 

In that case when you write a message it will be sent with your email as email_from, which is also the alias. Then when people respond to you, we detect they are responding to the alias and so we use the owner of the alias (also you) to write the message.

It's not necessary for the test but it's supposed to show why you would be responding to an alias in the first place, aside from people randomly deciding to edit the automatic email_to address when responding in their mail client.

----------------

Have added a comment to make that clearer but we can also remove it, it's really just to demonstrate the usecase.

Reply:
right, that makes more sense

I guess I can just remove it entirely then, yes